inc/functions.php

```diff
@@ -549,6 +549,64 @@ function my_date($format, $stamp=0, $offset="", $ty=1, $adodb=false)
 	return $date;
 }
 
+/**
+ * Get a mail handler instance, a MyBB's built-in SMTP / PHP mail hander or one created by a plugin.
+ * @param bool $use_buitlin Whether to use MyBB's built-in mail handler.
+ *
+ * @return object A MyBB's built-in mail handler or one created by plugin(s).
+ */
+function &get_mybb_mailhandler($use_buitlin = false)
+{
+	global $mybb, $plugins;
+	static $mybb_mailhandler;
+	static $mybb_mailhandler_builtin;
+
+	// If our built-in mail handler doesn't exist, create it.
+	if(!is_object($mybb_mailhandler_builtin))
```

Comment:
Why we are aggressive to create built-in object not checking the requirement passed by parameter?
We can create the static var later also while encountering use_buitlin true call for the very first time.

Reply:
Yeah, your way's better in terms of logic. I was writing that for better looking lol.